inductiva/molecules/scenarios/protein_solvation/protein_solvation.py

```diff
@@ -43,52 +43,27 @@ def __init__(
     def simulate(
             self,
             simulator: Simulator = GROMACS(),
-            working_dir: Optional[Path] = None,
+            output_dir: Optional[Path] = None,
             simulation_time: float = 10,  # ns
-            integrator: str = "md",
+            integrator: Literal["md", "sd", "bd"] = "md",
             nsteps_minim: int = 5000):
         """Simulate the solvation of a protein.
         Args:
-            working_dir: The working directory where the simulation 
-            will be executed.
-            simulation_time: The simulation time in ns. Default is 10 ns.
-            integrator: The integrator to use for the simulation. Default is md.
+            output_dir: The  output directory to save the results of the
+            simulation. 
+            simulation_time: The simulation time in ns. 
+            integrator: The integrator to use for the simulation. 
+            For more information about the integrator used in the simulation, 
+            consult the GROMACS official documentation at 
+            https://manual.gromacs.org/current/user-guide/mdp-options.html. 
             nsteps_minim: The number of steps to use for the energy minization. 
-            Default is 5000.
         """
         self.nsteps = int(
             simulation_time * 1e6 / 2
         )  # convert to fs and divide by the time step of the simulation (2 fs)
         self.integrator = integrator
         self.nsteps_minim = nsteps_minim
-        self.working_dir = self.setup_working_dir(working_dir, self.protein_pdb)
-        self.gen_config(simulator)
-        self.run_simulation(simulator)
-
-    def setup_working_dir(self, working_dir, protein_pdb):
-        """Setup the working directory for the simulation. 
-        For this scenario, the input and output directories are the same."""
-
-        # Create working_dir if it does not exist
-        if working_dir is None:
-            scenario_name_splitted = split_camel_case(self.__class__.__name__)
-            working_dir_prefix = "-".join(scenario_name_splitted).lower()
-            working_dir = get_timestamped_path(f"{working_dir_prefix}-output")
-
-        working_dir = resolve_path(working_dir)
-        if not os.path.exists(working_dir):
-            os.makedirs(working_dir)
-
-        # Copy protein pdb to working_dir
-        shutil.copy(protein_pdb,
-                    os.path.join(working_dir, os.path.basename(protein_pdb)))
-        # Copy template files to working_dir
-        shutil.copytree(os.path.join(self.template_dir),
-                        os.path.join(working_dir),
-                        dirs_exist_ok=True)
-        # Remove all files that have .jinja in the working_dir
-        remove_files_with_tag(working_dir, ".jinja")
-        return working_dir
+        return super().simulate(simulator, output_dir)
```

Comment:
Can we decide the type of hardware to run? Or GROMACs will consume everything it has available?

Just questioning, because David showed that not always is appropriate to use a lot of cores.